core/runtime/jsruntime.rs

```diff
@@ -1720,6 +1711,19 @@ impl JsRuntime {
       .map(|handle| v8::Local::new(tc_scope, handle))
       .expect("ModuleInfo not found");
     let mut status = module.get_status();
+    if status == v8::ModuleStatus::Evaluated {
+      let (sender, receiver) = oneshot::channel();
+      sender.send(Ok(())).unwrap();
+      return receiver;
+    }
+    if status == v8::ModuleStatus::Errored {
+      let (sender, receiver) = oneshot::channel();
+      let exception = module.get_exception();
+      sender
+        .send(exception_to_err_result(tc_scope, exception, false))
+        .unwrap();
+      return receiver;
+    }
```

Comment:
Can you explain once more why we need this? IMO this is a footgun during runtime, because it removes some invariants we had in the module loading system. If it is only needed for snapshotting then we should put it in a conditional block

Reply:
It's the other part of unifying `init_ops()` and `init_ops_and_esm()`. Isn't module evaluation supposed to be idempotent in the first place? It looks like that's what `Module::Evaluate()` does in v8.

Reply:
Okay, now I get it. Can you add comments here that remind that ES module loading is idempotent?

Reply:
Reverted this part since I had to move the check outside anyway (https://github.com/denoland/deno/pull/19519#discussion_r1233315331). It's still valid but unrelated to this PR.